package-system/AWSNativeSDK/docker_build_aws_sdk.sh

```diff
@@ -6,85 +6,207 @@
 # SPDX-License-Identifier: Apache-2.0 OR MIT
 #
 
-# Validate the src path
-src_path=src
-if [ ! -d $src_path ]
+
+# Validate the bld path input
+BUILD_FOLDER=${DOCKER_BUILD_PATH}
+if [ "${BUILD_FOLDER}" == "" ]
+then
+    echo "Missing required build target folder environment"
+    exit 1
+elif [ "${BUILD_FOLDER}" == "temp" ]
+then
+    echo "Build target folder environment cannot be 'temp'"
+    exit 1
+fi
+
+
+# Locate the dependent OpenSSL package
+OPENSSL_REGEX='(OpenSSL-([A-Za-z0-9\.\-]+)-(linux|linux-aarch64))'
+[[ $DOWNLOADED_PACKAGE_FOLDERS =~ $OPENSSL_REGEX ]]
+DEPENDENT_OPENSSL=${BASH_REMATCH[1]}
+
+if [ $DEPENDENT_OPENSSL == "" ]
+then
+    echo "Unable to detect dependent OpenSSL package"
+    exit 1
+fi
+DEPENDENT_OPENSSL_BASE=$WORKSPACE/temp/${DEPENDENT_OPENSSL}/OpenSSL
+if [ ! -d ${DEPENDENT_OPENSSL_BASE} ]
+then
+    echo "Unable to detect dependent OpenSSL package at ${DEPENDENT_OPENSSL_BASE}"
+    exit 1
+fi
+echo "Detected dependent OpenSSL package at ${DEPENDENT_OPENSSL_BASE}"
+
+
+# Locate the dependent ZLIB package
+OPENZLIB_REGEX='(zlib-([A-Za-z0-9\.\-]+)-(linux|linux-aarch64))'
+[[ $DOWNLOADED_PACKAGE_FOLDERS =~ $OPENZLIB_REGEX ]]
+DEPENDENT_ZLIB=${BASH_REMATCH[1]}
+
+if [ $DEPENDENT_ZLIB == "" ]
+then
+    echo "Unable to detect dependent zlib package"
+    exit 1
+fi
+DEPENDENT_ZLIB_BASE=$WORKSPACE/temp/${DEPENDENT_ZLIB}/zlib
+if [ ! -d ${DEPENDENT_ZLIB_BASE} ]
+then
+    echo "Unable to detect dependent zlib package at ${DEPENDENT_ZLIB_BASE}"
+    exit 1
+fi
+echo "Detected dependent zlib package at ${DEPENDENT_ZLIB_BASE}"
+
+
+# Prepare curl
+CURL_BASE=/data/workspace/curl
+CURL_SRC=${CURL_BASE}/src
+CURL_INSTALL=${CURL_BASE}/install
+if [ ! -d $CURL_BASE ]
+then
+    echo "Unable to find source curl library ($CURL_SRC) from this docker image"
+    exit 1
+fi
+
+# Build curl from source
+pushd $CURL_BASE/src
```

Comment:
I want to say in the past some users ran into issues when they tried to add other 3rdparty libraries that also depended on curl and they had fun version conflicts - I believe we had to resolve it by wrapping one version in a namespace.  Maybe we should vend a 3p curl package in the future that users could use.

Reply:
That is a problem in general with the 3rd party system, we don't have a true dependency resolving system so that has potential issues with other 3p libraries as well. Building non-monolithically is pretty safe from that issue, but this is an issue that will occur with monolithic builds.